src/relay/libp2p_transport_relay.erl

```diff
@@ -0,0 +1,73 @@
+-module(libp2p_transport_relay).
+
+-behavior(libp2p_transport).
+
+-export([
+    start_link/1
+    ,start_listener/2
+    ,match_addr/1
+    ,connect/5
+    ,reg/1
+]).
+
+%% ------------------------------------------------------------------
+%% libp2p_transport
+%% ------------------------------------------------------------------
+-spec start_link(ets:tab()) -> ignore.
+start_link(_TID) ->
+    ignore.
+
+-spec start_listener(pid(), string()) -> {error, unsupported}.
+start_listener(_Pid, _Addr) ->
+    {error, unsupported}.
+
+-spec match_addr(string()) -> {ok, string()} | false.
+match_addr(Addr) when is_list(Addr) ->
+    match_protocols(multiaddr:protocols(multiaddr:new(Addr))).
+
+-spec connect(pid(), string(), libp2p_swarm:connect_opts()
+              ,pos_integer(), ets:tab()) -> {ok, pid()} | {error, term()}.
+connect(_Pid, MAddr, Options, Timeout, TID) ->
+    {ok, {RAddress, AAddress}} = libp2p_relay:p2p_circuit(MAddr),
+    true = erlang:register(?MODULE:reg(AAddress), self()),
```

Comment:
Is this better than using the ETS TID?

Reply:
that's one comment that @madninja made also (but I forgot about it). Will use the ETS instead of `erlang:register`

Reply:
I remember @macpie mentioning that using a register would avoid having to add another monitor to clean up the ETS entry? 

Maybe that reason is gone now?

Reply:
If there's a good reason, I'm fine with it, but I view the process registry like the process dictionary; something that should be used with care.